SQRLDotNetClientUI/Platform/Linux/SystemEventNotifier.cs

```diff
@@ -0,0 +1,121 @@
+        public event EventHandler<SystemEventArgs> SessionLock;
+        public event EventHandler<SystemEventArgs> ShutdownOrRestart;
+#pragma warning restore 67
+        private IntPtr Display;
+        private X11.XScreenSaverInfo xsi;
+
+        private DateTime lastCheck;
+
+        public SystemEventNotifier(int maxIdleSeconds = 60 * 15)
+        {
+            
+            _cts = new CancellationTokenSource();
+            _ct = _cts.Token;
+
+            this._maxIdleSeconds = maxIdleSeconds;
+            Display = X11.Xlib.XOpenDisplay(null);
+            xsi =  new X11.XScreenSaverInfo();
+            lastCheck = DateTime.Now;
+            _pollTask = new Task(() =>
+            {
+                Log.Information("SystemEventNotifier polling task started");
+                
+                while (!_ct.IsCancellationRequested)
+                {
+                    // Check system idle time
+                    X11.Xlib.XScreenSaverQueryInfo(Display, X11.Xlib.XRootWindow(Display, 0), ref xsi);
+                    
+                    Log.Debug("Idle time: {IdleTime}", xsi.idle.ToString());
+                    if (xsi.idle > (ulong)_maxIdleSeconds)
+                    {
+                        if (!_idleDetected)
+                        {
+                            Idle?.Invoke(this, new SystemEventArgs("Idle Timeout"));
+                            Log.Information("Detected idle timeout");
+                            _idleDetected = true;
+                        }
+                    }
+                    else
+                    {
+                        if (_idleDetected)
+                            _idleDetected = false;
+                    }
+
+                    if ((ScreenSaverState)xsi.state == ScreenSaverState.ScreenSaverOn)
+                    {
+                        if (!_screenSaver)
+                        {
+                            Screensaver?.Invoke(this, new SystemEventArgs("Screen Saver"));
+                            Log.Information("Detected Screen Saver On");
+                            _screenSaver = true;
+                        }
+                    }
+                    else if (_screenSaver)
+                        _screenSaver =false;
+
+
+                    //Since we can't check for Lock, Logout or Restart we are going to
+                    //Do a bit of  a hack and assume that if our loop took longer than triple the standard time to run
+                    //then we want to go ahead and clear the quick pass;
+                    if ((DateTime.Now -lastCheck).TotalSeconds > (POLL_INTERVAL/1000) * 3)
+                    {
+                        Log.Information("Detected an Anommaly System Loop took > 3X to run, clearing quick pass");
+                        Standby?.Invoke(this, new SystemEventArgs("System Hiccup Clearing QuickPass Just in Case"));
+                    }
+                    lastCheck = DateTime.Now;
```

Comment:
@alexhauser  this is the bit of code I'm using to check if the POLL is taking 3X> than normal... Thoughts on this approach? Do we care? Do we need it? Do we want it?